internal/pkg/deploy/cloudformation/project.go

```diff
@@ -439,3 +439,45 @@ func (cf CloudFormation) waitForStackSetOperation(stackSetName, operationID stri
 		time.Sleep(3 * time.Second)
 	}
 }
+
+// DeleteProject deletes all project specific StackSet and Stack resources.
+func (cf CloudFormation) DeleteProject(name string, accounts, regions []string) error {
+	if _, err := cf.client.DeleteStackInstances(&cloudformation.DeleteStackInstancesInput{
+		Accounts:     aws.StringSlice(accounts),
+		RetainStacks: aws.Bool(false),
+		Regions:      aws.StringSlice(regions),
+		StackSetName: aws.String(fmt.Sprintf("%s-infrastructure", name)),
+	}); err != nil {
+		return err
+	}
+
+	maxAttempts := 10
+	for maxAttempts > 0 {
+		_, err := cf.client.DeleteStackSet(&cloudformation.DeleteStackSetInput{
+			StackSetName: aws.String(fmt.Sprintf("%s-infrastructure", name)),
+		})
+
+		if err != nil {
+			awserr, ok := err.(awserr.Error)
+			if !ok {
+				return err
+			}
+
+			if awserr.Code() == cloudformation.ErrCodeOperationInProgressException {
+				maxAttempts--
+				time.Sleep(30 * time.Second)
```

Comment:
Do we need to loop to ensure the stackset is deleted because it needs to be gone before deleting the `infrastructure-roles` stack?

If so can we add a comment explaining it?

Reply:
The looping here is because there's no `Wait` convenience function the SDK provides for waiting on StackSet Instances to delete. The `DeleteStackSet` will return the `ErrCodeOperationInProgressException` while instances are still actively being deleted. It's not related to the `roles` stack.